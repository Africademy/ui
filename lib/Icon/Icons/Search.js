"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _createIconComponent = _interopRequireDefault(require("../utils/createIconComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
var SearchIcon = (0, _createIconComponent.default)({
  content: _react.default.createElement("g", null, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M9.5 10a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7zm0 2a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11z"
  }), _react.default.createElement("path", {
    d: "M5.95 9l1.414 1.414-4.243 4.243a1 1 0 0 1-1.414-1.414L5.95 9z"
  }))
});
SearchIcon.displayName = 'SearchIcon';
var _default = SearchIcon;
exports.default = _default;