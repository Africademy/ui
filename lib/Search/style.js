"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchInput = exports.SearchWrapper = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    && {\n      border: 0;\n      border-bottom:  1px solid #ecf0f3;\n      border-radius: 4px;\n      box-shadow: none;\n      width: 100%;\n      padding: 0 7px;\n      box-sizing: border-box;\n      resize: none;\n      font-size: 14px;\n      height: 35px;\n      &:focus {\n        outline: none;\n        border-color: transparent;\n        box-shadow: none;\n        border-bottom: 1px solid #b8b8b8;\n        border-bottom-left-radius: 0px;\n        border-bottom-right-radius: 0px;\n      }\n    }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  height:  auto;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var SearchWrapper = _styledComponents.default.div(_templateObject());

exports.SearchWrapper = SearchWrapper;

var SearchInput = _styledComponents.default.input(_templateObject2());

exports.SearchInput = SearchInput;