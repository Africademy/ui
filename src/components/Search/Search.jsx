import React from 'react';
import PropTypes from 'prop-types';
import { SearchInput, SearchWrapper } from './style';

const map = {
  close: 'esc',
  addItem: 'enter',
  moveUp: 'up',
  moveDown: 'down',
};

/** Search input that filters the Select items and adds keyboard navigation */
export default class Search extends React.Component {
  state = {
    search: '',
  };

  onChange = (event) => {
    console.info(event);
    const { onChange } = this.props;
    const search = event.target.value;

    onChange(search);
    this.setState({
      search,
    });
  }

  getHotkeyHandlers = () => {
    const {
      onAddItem, onClose,
    } = this.props;
    return {
      addItem: () => {
        onAddItem();
      },
      close: () => {
        onClose();
      },
    };
  };

  render() {
    const {
      shortcutsEnabled, placeholder,
    } = this.props;

    const { search } = this.state;

    // adding a small delay to make sure the rendering is complete
    if (this.inputRef) setTimeout(() => this.inputRef.focus(), 10);

    return (
      <SearchWrapper keyMap={map} handlers={shortcutsEnabled ? this.getHotkeyHandlers() : undefined}>
        <SearchInput
          placeholder={placeholder}
          type="text"
          value={search}
          ref={inputRef => this.inputRef = inputRef}
          onChange={event => this.onChange(event)}
        />
      </SearchWrapper>
    );
  }
}

Search.propTypes = {
  /** If false don't enable keyboard navigation */
  shortcutsEnabled: PropTypes.bool,

  /** Search placeholder */
  placeholder: PropTypes.string,

  /** Function to call on search input change */
  onChange: PropTypes.func.isRequired,

  /** Function to call on Enter click */
  onAddItem: PropTypes.func.isRequired,

  /** Function to call on Esc click */
  onClose: PropTypes.func.isRequired,
};

Search.defaultProps = {
  shortcutsEnabled: true,
  placeholder: 'Search',
};
