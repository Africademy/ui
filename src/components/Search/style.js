import style from 'styled-components';
import { HotKeys } from 'react-hotkeys';

export const SearchWrapper = style(HotKeys)`
  height:  auto;
`;

export const SearchInput = style.input`
    border: 0;
    border-bottom:  1px solid #ecf0f3;
    border-radius: 4px;
    box-shadow: none;
    width: 100%;
    padding: 0 7px;
    box-sizing: border-box;
    resize: none;
    font-size: 14px;
    height: 35px;
    &:focus {
      outline: none;
    }
`;
