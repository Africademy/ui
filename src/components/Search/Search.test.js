/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from './Search';

Enzyme.configure({ adapter: new Adapter() });

describe('SomeComponent component', () => {
  it('onChange: should update state', () => {
    const wrapper = shallow(<Search onChange={() => true} onMoveUp={() => true} onMoveDown={() => true} onAddItem={() => true} onClose={() => true} />);
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 'Test',
      },
    };
    instance.onChange(event);
    expect(wrapper.state().search).toBe('Test');
  });

  /** TODO: Fix this test. We need to check that getHotkeyHandles is called on render */
  it('should call MoweDown on arrow down', () => {
    const wrapper = mount(<Search hasSearch onChange={() => true} onMoveUp={() => true} onMoveDown={() => true} onAddItem={() => true} onClose={() => true} />);
    const spy = jest.spyOn(wrapper.instance(), 'getHotkeyHandlers');
    wrapper.instance().getHotkeyHandlers();
    expect(spy).toHaveBeenCalled();
  });


  it('getHotkeyHandlers should return the handler', () => {
    const wrapper = mount(<Search hasSearch onChange={() => true} onMoveUp={() => true} onMoveDown={() => true} onAddItem={() => true} onClose={() => true} />);
    const mockFunction = wrapper.instance().getHotkeyHandlers();
    expect(mockFunction).toMatchSnapshot();
  });
});
