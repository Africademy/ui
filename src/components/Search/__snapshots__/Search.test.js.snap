// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent component getHotkeyHandlers should return the handler 1`] = `
Object {
  "addItem": [Function],
  "close": [Function],
}
`;
